osbs/tekton.py

```diff
@@ -0,0 +1,711 @@
+from requests.utils import guess_json_utf
+
+
+
+
+logger = logging.getLogger(__name__)
+
+
+# Retry each connection attempt after 30 seconds, for a maximum of 10 times
+WATCH_RETRY_SECS = 30
+WATCH_RETRY = 10
+MAX_BAD_RESPONSES = WATCH_RETRY // 3
+# Give up after 12 hours
+WAIT_RETRY_HOURS = 12
+WAIT_RETRY = WAIT_RETRY_HOURS * 3600 // (WATCH_RETRY_SECS * WATCH_RETRY)
+
+API_VERSION = "tekton.dev/v1beta1"
+
+
+def check_response(response, log_level=logging.ERROR):
+    if response.status_code not in (requests.status_codes.codes.ok, requests.status_codes.codes.created):
+        if hasattr(response, 'content'):
+            content = response.content
+        else:
+            content = b''.join(response.iter_lines())
+
+        logger.log(log_level, "[%d] %s", response.status_code, content)
+        raise OsbsResponseException(message=content, status_code=response.status_code)
+
+class Openshift(object):
+    def __init__(self, openshift_api_url, openshift_oauth_url,
+                 k8s_api_url=None,
+                 verbose=False, username=None, password=None, use_kerberos=False,
+                 kerberos_keytab=None, kerberos_principal=None, kerberos_ccache=None,
+                 client_cert=None, client_key=None, verify_ssl=True, use_auth=None,
+                 token=None, namespace=DEFAULT_NAMESPACE):
+        self.os_api_url = openshift_api_url
+        self.k8s_api_url = k8s_api_url
+        self._os_oauth_url = openshift_oauth_url
+        self.namespace = namespace
+        self.verbose = verbose
+        self.verify_ssl = verify_ssl
+        self._con = HttpSession(verbose=self.verbose)
+        self.retries_enabled = True
+
+        # auth stuff
+        self.use_kerberos = use_kerberos
+        self.username = username
+        self.password = password
+        self.client_cert = client_cert
+        self.client_key = client_key
+        self.kerberos_keytab = kerberos_keytab
+        self.kerberos_principal = kerberos_principal
+        self.kerberos_ccache = kerberos_ccache
+        self.token = token
+
+        self.ca = None
+        auth_credentials_provided = bool(use_kerberos or
+                                         token or
+                                         (username and password))
+        if use_auth is None:
+            self.use_auth = auth_credentials_provided
+            if not self.use_auth:
+                # Are we running inside a pod? If so, we will have a
+                # token available which can be used for authentication
+                self.use_auth = self.can_use_serviceaccount_token()
+        else:
+            self.use_auth = use_auth
+            if not auth_credentials_provided:
+                # We've been told to use authentication but no
+                # credentials have been given. See if we're running
+                # inside a pod, and if so use the provided token.
+                self.can_use_serviceaccount_token()
```

Comment:
Is it ok to just discard the return value here? Shouldn't this raise an exception if `use_auth and not auth_credentials_provided and not self.can_use_serviceaccount_token()`?